modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java

```diff
@@ -153,45 +169,80 @@ SynonymMap buildSynonyms(Analyzer analyzer, ReaderWithOrigin rules) {
     }
 
     protected ReaderWithOrigin getRulesFromSettings(Environment env, IndexCreationContext context) {
-        if (settings.getAsList("synonyms", null) != null) {
-            List<String> rulesList = Analysis.getWordList(env, settings, "synonyms");
-            StringBuilder sb = new StringBuilder();
-            for (String line : rulesList) {
-                sb.append(line).append(System.lineSeparator());
+        return switch (synonymsSource) {
+            case INLINE -> {
+                List<String> rulesList = Analysis.getWordList(env, settings, SynonymsSource.INLINE.getSettingName());
+                StringBuilder sb = new StringBuilder();
+                for (String line : rulesList) {
+                    sb.append(line).append(System.lineSeparator());
+                }
+                yield new ReaderWithOrigin(new StringReader(sb.toString()), "'" + name() + "' analyzer settings");
             }
-            return new ReaderWithOrigin(new StringReader(sb.toString()), "'" + name() + "' analyzer settings");
-        } else if (settings.get("synonyms_set") != null) {
-            if (analysisMode != AnalysisMode.SEARCH_TIME) {
-                throw new IllegalArgumentException(
-                    "Can't apply [synonyms_set]! " + "Loading synonyms from index is supported only for search time synonyms!"
-                );
+            case INDEX -> {
+                if (analysisMode != AnalysisMode.SEARCH_TIME) {
+                    throw new IllegalArgumentException(
+                        "Can't apply ["
+                            + SynonymsSource.INDEX.getSettingName()
+                            + "]! Loading synonyms from index is supported only for search time synonyms!"
+                    );
+                }
+                String synonymsSet = settings.get(SynonymsSource.INDEX.getSettingName(), null);
+                // provide empty synonyms on index creation and index metadata checks to ensure that we
+                // don't block a master thread
+                ReaderWithOrigin reader;
+                if (context != IndexCreationContext.RELOAD_ANALYZERS) {
+                    reader = new ReaderWithOrigin(
+                        new StringReader(""),
+                        "fake empty [" + synonymsSet + "] synonyms_set in .synonyms index",
+                        synonymsSet
+                    );
```

Comment:
Changed the filter to initialize with an empty synonyms set instead of a fake rule. IMO this is better because now there is zero chance the user hits the fake rule in between init and when analyzers are reloaded. This empty synonyms set handling is tested in `SynonymsApiIT#testLoadIndexWithEmptySynonymsSet`.